src/main/resources/mappings/workflow-state.json

```diff
@@ -31,7 +31,15 @@
       "type": "object"
     },
     "resources_created": {
-      "type": "object"
+      "type": "nested",
+      "properties": {
+        "workflow_step_name": {
+          "type": "keyword"
+        },
+        "resource_id": {
+          "type": "keyword"
+        }
+      }
```

Comment:
can we persist resource types as well? Frontend mocks currently have this included, e.g., "index", "model/connector"

Reply:
We could do this, but we would need some better mapping then what we have now: discussed some of this here: https://github.com/opensearch-project/opensearch-ai-flow-framework/issues/136

right now we just have the step name which means we need some mapping between step name and resources itself or to that info somewhere earlier.

Reply:
I see. I think it will be important to portray to the user a readable list of created resources to provide good transparency as to what this plugin is doing behind the scenes. The user should easily be able to see all associated connectors, knn indices, etc. that they've created so they can keep track of things. Just a ref to the workflow step name doesn't seem sufficient.

I'll add a comment in #136 since this is still a WIP. No blocker for this PR